blob/gcsblob/gcsblob.go

```diff
@@ -611,6 +737,10 @@ func (b *bucket) SignedURL(ctx context.Context, key string, dopts *driver.Signed
 		PrivateKey:     b.opts.PrivateKey,
 		SignBytes:      b.opts.SignBytes,
 	}
+	otherSignerAvailable := len(opts.PrivateKey) >= 1 || opts.SignBytes != nil
```

Comment:
This "if" is confusing me. Can we just check at bucket creation time that only one of SignBytes/MakeSignBytes is provided, and then simplify this to:

sb := opts.SignBytes
if opts.MakeSignBytes != nil {
  sb = opts.MakeSignBytes(ctx)
}
...

?

Reply:
1. Both need to be checked, `SignBytes` and `PrivateKey`, so we don't involve a `MakeSignBytes` that will initialize something.
2. You've probably meant this exemplary, but the contract is that `MakeSignBytes` won't overwrite a `SignBytes` – the latter is prefered as it's supposedly without context, and apparently without calling any remote API.